### PR TITLE
common: removed Retry in favor of github.com/avast/retry-go

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch
run scale-tests

**Comment by lebauce** [comment]
run scale-tests

**Comment by lebauce** [comment]
run scale-tests

**Comment by lebauce** [comment]
run go-fmt
run unit-tests
run compile-tests
run functional-tests-backend-orientdb
run functional-hw-tests
run cdd-overview-tests
run k8s-tests

File: tests/api_test.go [hunk 1]
```diff
@@ -19,20 +19,20 @@ package tests
...
 	shttp "github.com/skydive-project/skydive/http"
 )
 
 func getCrudClient() (c *shttp.CrudClient, err error) {
-	common.Retry(func() error {
+	retry.Do(func() error {
 		c, err = client.NewCrudClientFromConfig(&shttp.AuthenticationOpts{})
 		return err
-	}, 10, time.Second)
+	})
```
Comment (nplanel) [inline tests/api_test.go]: ... , Delay(10*time.Second))

default delay between retries in retry.Do() is 100ms
So there are many place to fix ...


Why changing everything for no impact, except one more dependency ?

Comment (lebauce) [reply tests/api_test.go]: 10 retries with a delay of 1 second was chosen without any real reason and this is the case for almost every common.Retry (even more for the tests where we basically copied/pasted the same 10 * 1 second line)

We are trying to remove as much as possible of the ugly code we have in common/ and the retry-go library does exactly the same (and more) in a more elegant way so I don't see any reason to keep.

I took the opportunity of this PR to look at every common.Retry and see if the number of retry, the delay and the delay mechanism (linear or exponential).

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch
run unit-tests

**Comment by lebauce** [comment]
run scale-tests

File: tests/api_test.go (hunk 1 above)
Comment (lebauce) [reply tests/api_test.go]: For the Retry used in tests, if they pass, that's probably enough.

For the others, we should be careful. I tried to be but it's definitely I may have missed 1 or 2

**Pull request merged**